v2/spanner-to-sourcedb/src/test/java/com/google/cloud/teleport/v2/templates/SpannerToCassandraSourceDbIT.java

```diff
@@ -0,0 +1,929 @@
+    spannerAccessor
+        .getDatabaseClient()
+        .readWriteTransaction(
+            Options.tag("txBy=forwardMigration"),
+            Options.priority(spannerConfig.getRpcPriority().get()))
+        .run(
+            (TransactionRunner.TransactionCallable<Void>)
+                transaction -> {
+                  Mutation m3 =
+                      Mutation.newInsertOrUpdateBuilder("users")
+                          .set("id")
+                          .to(3)
+                          .set("full_name")
+                          .to("GG")
+                          .set("from")
+                          .to("BB")
+                          .build();
+                  transaction.buffer(m3);
+                  return null;
+                });
+  }
+
+  /**
+   * Asserts that a basic row exists in the Cassandra database.
+   *
+   * <p>This method performs the following steps:
+   *
+   * <ul>
+   *   <li>Waits for the condition that ensures one row exists in the Cassandra table {@code
+   *       USER_TABLE}.
+   *   <li>Retrieves and logs rows from the Cassandra table.
+   *   <li>Checks if exactly one row is present in the table.
+   *   <li>Verifies that the row contains expected values for columns: {@code id}, {@code
+   *       full_name}, and {@code from}.
+   * </ul>
+   *
+   * @throws InterruptedException if the thread is interrupted while waiting for the row count
+   *     condition.
+   * @throws RuntimeException if reading from the Cassandra table fails.
+   */
+  private void assertBasicRowInCassandraDB() throws InterruptedException {
+    PipelineOperator.Result result =
+        pipelineOperator()
+            .waitForCondition(
+                createConfig(jobInfo, Duration.ofMinutes(10)), () -> getRowCount(USER_TABLE) == 1);
+    assertThatResult(result).meetsConditions();
+    Iterable<Row> rows;
+    try {
+      LOG.info("Reading from Cassandra table: {}", USER_TABLE);
+      rows = cassandraResourceManager.readTable(USER_TABLE);
+      LOG.info("Cassandra Rows: {}", rows.toString());
+    } catch (Exception e) {
+      throw new RuntimeException("Failed to read from Cassandra table: " + USER_TABLE, e);
+    }
+
+    assertThat(rows).hasSize(1);
+
+    Row row = rows.iterator().next();
+    LOG.info("Cassandra Row to Assert: {}", row.toString());
+    assertThat(row.getInt("id")).isEqualTo(1);
+    assertThat(row.getString("full_name")).isEqualTo("A");
+    assertThat(row.getString("from")).isEqualTo("B");
+  }
+
+  /**
+   * Writes a row containing all supported data types into the Spanner database.
+   *
+   * <p>This method creates and inserts a row into the {@code ALL_DATA_TYPES_TABLE} with various
+   * data types, including text, numerical, date/time, boolean, byte arrays, lists, sets, and maps.
+   * The values are set explicitly to ensure compatibility with Spanner's schema.
+   *
+   * <p>The following data types are included:
+   *
+   * <ul>
```

Comment:
you can remove these verbose comments, the methods are self explainatory. Same comment for other places